packages/marshal/src/encodePassable.js

```diff
@@ -355,40 +484,45 @@ export const makeEncodePassable = (encodeOptions = {}) => {
         return encodeBigInt(passable);
       }
       case 'remotable': {
-        const result = encodeRemotable(passable, encodePassable);
+        const result = encodeRemotable(passable, innerEncode);
         result.startsWith('r') ||
           Fail`internal: Remotable encoding must start with "r": ${result}`;
         return result;
       }
       case 'error': {
-        const result = encodeError(passable, encodePassable);
+        const result = encodeError(passable, innerEncode);
         result.startsWith('!') ||
           Fail`internal: Error encoding must start with "!": ${result}`;
         return result;
       }
       case 'promise': {
-        const result = encodePromise(passable, encodePassable);
+        const result = encodePromise(passable, innerEncode);
         result.startsWith('?') ||
           Fail`internal: Promise encoding must start with "?": ${result}`;
         return result;
       }
       case 'symbol': {
-        return `y${nameForPassableSymbol(passable)}`;
+        const encName = encodeString(nameForPassableSymbol(passable));
+        return `y${encName.slice(1)}`;
       }
       case 'copyArray': {
-        return encodeArray(passable, encodePassable);
+        return encodeArray(passable, innerEncode);
       }
       case 'copyRecord': {
-        return encodeRecord(passable, encodePassable);
+        return encodeRecord(passable, encodeArray, innerEncode);
       }
       case 'tagged': {
-        return encodeTagged(passable, encodePassable);
+        return encodeTagged(passable, encodeArray, innerEncode);
       }
       default: {
         throw Fail`a ${q(passStyle)} cannot be used as a collection passable`;
       }
     }
   };
+  const encodePassable = xxx
+    ? // A leading "#" indicates the v2 encoding (with escaping in strings rather than arrays).
+      passable => `#${innerEncode(passable)}`
+    : innerEncode;
```

Comment:
This leading `#` to indicate the new encoding is analogous to indication of smallcaps in a CapData `body`, but I'm open to other ideas.

Reply:
Should we pick a different indication? It may be worth observing that a CapData smallcaps body always follows the initial "#" with JSON text, while this marshal encoding always follows it with a character that cannot start JSON text (although the old encoding uses "[" for arrays, which would require further lookahead if we wanted that level of compatibility).

Reply:
> Should we pick a different indication?

Yes, I think so. There are plenty of ascii candidate characters left ;)

Reply:
OK, looking at https://github.com/endojs/endo/blob/7f1f3319acbd06fa28c08aa12f276ce3532ef6f0/packages/marshal/src/encodePassable.js#L494-L508 and also keeping in mind https://github.com/endojs/endo/blob/7f1f3319acbd06fa28c08aa12f276ce3532ef6f0/packages/marshal/src/encodeToSmallcaps.js#L47-L59 to avoid gratuitous divergence, how would you feel about prefixing with `~`?

Reply:
> how would you feel about prefixing with `~` ?

Good.